projects/demo/src/app/forms/dynamic-controls/dynamic-controls.demo.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ChangeDetectorRef, Component } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { filter, takeUntil, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'dynamic-controls-demo',
+  templateUrl: './dynamic-controls.demo.html',
+})
+export class DynamicControlsDemo {
+  destroyed = new Subject();
+  form = new FormGroup<any>({
+    platform: new FormControl(),
+    scenario: new FormControl('one'),
+  });
+
+  constructor(private cd: ChangeDetectorRef) {}
+
+  ngOnInit() {
+    this.form
+      .get('platform')
+      ?.valueChanges.pipe(
+        takeUntil(this.destroyed),
```

Comment:
This must be last to ensure all inner subscriptions are unsubscribed.

Reply:
If it's last then it will go trough again and then complete the inner subscriptions but if it's first then it completes and doesn't pass any values to the next pipe operators.
Created an example where it can be seen: https://stackblitz.com/edit/typescript-3ckpfc?devToolsHeight=33&file=index.ts

Reply:
That is because all of the value are emitted synchronously. This is a bad test.

If you emit the values in different frames using `setTimeout`:
- With `takeUntil` first, it will continue to emit values after destroy and the inner subscription in the `mapCopy` operator is not unsubscribed. Both of these are memory leaks.
- With `takeUntil` last, both issues are solved.

https://stackblitz.com/edit/typescript-nqfkh9?file=index.ts

It might be better to just call `unsubscribe` explicitly in `ngOnDestroy` so none of this is an issue.

Reply:
I am looking into this further. I am worried out the inner subscriptions of subsequent operators not being unsubscribed.

Reply:
The inner subscriptions of subsequent operators are not unsubscribed if `takeUntil` is first.

Here is an example with four demos, `takeUntil` first and last with values emitted synchronously and asynchronously for both:

https://stackblitz.com/edit/typescript-nqfkh9?devToolsHeight=33&file=index.ts

With `takeUntil` first: While values are not processed after destroy synchronously or asynchronously, the inner subscription in the `mapCopy` operator is never unsubscribed. This is a memory leak.

With `takeUntil` last: Synchronously-emitted values are processed after destroy, but the inner subscription in the `mapCopy` operator is always unsubscribed. This is not a memory leak because once the next frame hits, everything is done.

It might be better to just call `unsubscribe` explicitly in `ngOnDestroy` so none of this is an issue.

Reply:
The inner subscriptions of subsequent operators are not unsubscribed if `takeUntil` is first.

Here is an example with four demos, `takeUntil` first and last with values emitted synchronously and asynchronously for both:

https://stackblitz.com/edit/typescript-nqfkh9?devToolsHeight=33&file=index.ts

With `takeUntil` first: While values are not processed after destroy synchronously or asynchronously, the inner subscription in the `destination` operator is never unsubscribed. This is a memory leak.

With `takeUntil` last: While values are processed after destroy if (and only if) emitted synchronously, the inner subscription in the `destination` operator is always unsubscribed. This is not a memory leak because everything is done once the next frame hits. I think it's likely for a source event to be emitted synchronously in the same frame when the component is being destroyed, so that probably a non-issue as well.

All that said, it might be better to just call `unsubscribe` explicitly in `ngOnDestroy` so none of this is an issue at all.

Reply:
The inner subscriptions of subsequent operators are not unsubscribed if `takeUntil` is first.

Here is an example with four demos, `takeUntil` first and last with values emitted synchronously and asynchronously for both:

https://stackblitz.com/edit/typescript-nqfkh9?devToolsHeight=33&file=index.ts

With `takeUntil` first: While values are not processed after destroy synchronously or asynchronously, the inner subscription in the `destination` operator is never unsubscribed. This is a memory leak.

With `takeUntil` last: While values are processed after destroy if (and only if) emitted synchronously, the inner subscription in the `destination` operator is always unsubscribed. This is not a memory leak because everything is done once the next frame hits. I think it's unlikely for a source event to be emitted synchronously in the same frame when the component is being destroyed, so that probably a non-issue as well.

All that said, it might be better to just call `unsubscribe` explicitly in `ngOnDestroy` so none of this is an issue at all.

Reply:
https://github.com/cartant/eslint-plugin-rxjs/blob/main/docs/rules/no-unsafe-takeuntil.md

Reply:
Thanks for digging further! Moved it to be last

Reply:
Thanks!